doc/languages-frameworks/haskell.section.md

```diff
@@ -1,7 +1,877 @@
-# Haskell {#haskell}
+# Haskell {#sec-haskell}
 
-The documentation for the Haskell infrastructure is published at
-<https://haskell4nix.readthedocs.io/>. The source code for that
-site lives in the `doc/` sub-directory of the
-[`cabal2nix` Git repository](https://github.com/NixOS/cabal2nix)
-and changes can be submitted there.
+The Haskell infrastructure in nixpkgs has two main purposes: The primary purpose
+is to provide a Haskell compiler and build tools as well as infrastructure for
+packaging Haskell-based packages.
+
+The secondary purpose is to provide support for Haskell development environment
+including prebuilt Haskell libraries. However, in this area sacrifices have been
+made due to self-imposed restrictions in nixpkgs, to lessen the maintenance
+effort and improve performance. Therefore, it may be advantageous to use an
+alternative to the Haskell infrastructure in nixpkgs for development
+environments in some cases. The main limitations are that we only provide
+first-class support for the default compiler (currently GHC 8.10.7) and usually
```

Comment:
```suggestion
first-class support for the default compiler (currently GHC 9.0.2) and usually
```

or maybe we remove this line? I feel like it will bite us over and over?

Reply:
Thanks, updated references to 8.10.7. I think it's fine once we've merged since I at least always do a grep over the repo to find references to the old GHC version; e.g. module docs already reference ghc vresions today, so there's always something to update.